File: AspNetMvcUnleashed/0672329980_source/Chapter14/CS/MvcApplication1/MvcApplication1/Controllers/ServerValidateController.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class ServerValidateController : Controller
    {
        private GuestBookDBEntities _entities = new GuestBookDBEntities();

        // GET: /GuestBook/

        public ActionResult Index()
        {
            return View(_entities.GuestSet.ToList());
        }

        // POST: /GuestBook/Create

        public ActionResult Create([Bind(Exclude="Id")]Guest guestToCreate)
        {
            if (guestToCreate.Name.Trim().Length == 0)
                ModelState.AddModelError("Name", "Name is required.");
            if (guestToCreate.Message.Trim().Length == 0)
                ModelState.AddModelError("Message", "Message is required.");

            if (ModelState.IsValid)
            {
                _entities.AddToGuestSet(guestToCreate);
                _entities.SaveChanges();
            }

            return PartialView("GuestBook", _entities.GuestSet.ToList());
        }
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter14/VB/MvcApplication1/MvcApplication1/Controllers/ServerValidateController.vb
Public Class ServerValidateController
    Inherits Controller

    Private _entities As New GuestBookDBEntities()

    ' GET: /GuestBook/

    Public Function Index() As ActionResult
        Return View(_entities.GuestSet.ToList())
    End Function

    ' POST: /GuestBook/Create

    Public Function Create(<Bind(Exclude:="Id")> ByVal guestToCreate As Guest) As ActionResult
        If guestToCreate.Name.Trim().Length = 0 Then
            ModelState.AddModelError("Name", "Name is required.")
        End If
        If guestToCreate.Message.Trim().Length = 0 Then
            ModelState.AddModelError("Message", "Message is required.")
        End If

        If ModelState.IsValid Then
            _entities.AddToGuestSet(guestToCreate)
            _entities.SaveChanges()
        End If

        Return PartialView("GuestBook", _entities.GuestSet.ToList())
    End Function
End Class
